### ISSUE TITLE
`IBook.changeMaterialItem` not working

### ISSUE THREAD

**Issue by Krutoy242** [Project-RT/RandomTweaker#66]
I used  `mods.randomtweaker.tconstruct.IBook.changeMaterialItem('xu_magical_wood', <extrautils2:decorativesolidwood:1>);` to change item representation in book.
This what i see in logs:

```less
[INITIALIZATION][CLIENT][INFO] [crafttweaker | SIDE_CLIENT]: Loading Script: {[0:crafttweaker]: mods\randomtweaker.zs}
[INITIALIZATION][CLIENT][INFO] Changing material fusewood to item <botania:shimmerwoodplanks>
[INITIALIZATION][CLIENT][INFO] Changing material xu_magical_wood to item <extrautils2:decorativesolidwood:1>
```

This what i see in book. Its didn't changed =(

![image](https://github.com/Project-RT/RandomTweaker/assets/936618/cfdcbb6f-6c05-4a2b-843a-bc78a635473f)

**Comment by ikexing-cn** [comment]
The implementation within the `IAction` of`changeMaterialItem` method only checks for the existence of a key related to the material in the `HashMap`. 

`If no duplicates are found, it logs a success message`.

But it doesn't verify the legitimacy of the material itself. 
So, I think you need to manually ensure that the string you input is a valid material.

**Comment by Krutoy242** [comment]
I checked - Shard of this material have `xu_magical_wood` in its NBT tag. Also, `xu_magical_wood` key is used in my other tweak mods like TweakerConstruct.
Also, i tried this tweak with another material `fusewood` and its not working too.

There is debug.log file and its have this line:
```less
[Client thread/INFO] [toolprogression]: TiC Material registered: xu_magical_wood
```

[debug.log](https://github.com/Project-RT/RandomTweaker/files/13176165/debug.log)

**Comment by ikexing-cn** [comment]
![image](https://github.com/Project-RT/RandomTweaker/assets/75018198/43e35181-89a4-4952-aa14-5b44c9766aef)

hmmm, it works in my test, I believe it could be an issue related to compatibility with other expansion mods for TC.
It's quite challenging to debug. Could you please launch the game and then upload the ASMDebug.class file to the .minecraft directory? Additionally, I would need the latest.log file to help me understand what's happening.

[RandomTweaker-1.4.0.debug.zip](https://github.com/Project-RT/RandomTweaker/files/13185072/RandomTweaker-1.4.0.debug.zip)

**Comment by ikexing-cn** [comment]
![image](https://github.com/Project-RT/RandomTweaker/assets/75018198/43e35181-89a4-4952-aa14-5b44c9766aef)

hmmm, it works in my tests, I believe it could be an issue related to compatibility with other expansion mods for TC.
It's quite challenging to debug. Using the latest built mod, after the game has started, you can search for the `ASMDebug.class` file in the `.minecraft` directory and then upload it. Additionally, I would need the` latest.log `file to help me understand what's happening.

[RandomTweaker-latest.debug.zip](https://github.com/Project-RT/RandomTweaker/files/13185072/RandomTweaker-1.4.0.debug.zip)

**Comment by Krutoy242** [comment]
Thank you! I will try this later.

**Comment by Krutoy242** [comment]
There you go!

[ASMDebug.zip](https://github.com/Project-RT/RandomTweaker/files/13279058/ASMDebug.zip)

[debug.log](https://github.com/Project-RT/RandomTweaker/files/13279060/debug.log)

**Comment by ikexing-cn** [comment]
All of these are expected result, sorry I couldn't offer any more help.

**Comment by Krutoy242** [comment]
I will try to find other mod that incompatible then.

**Comment by ikexing-cn** [comment]
Fixed, [#1397ec6](https://github.com/Project-RT/RandomTweaker/commit/1397ec6a4f67ebbe8ac802d0a026df3a4dc04628)
